jax/experimental/jax2tf/tests/jax2tf_limitations.py

```diff
@@ -825,29 +825,36 @@ def custom_assert(tst, result_jax, result_tf, *, args, tol, err_msg):
   @classmethod
   def igammac(cls, harness: test_harnesses.Harness):
     dtype = harness.dtype
-
     # igammac is not defined when the first argument is <=0
-    def custom_assert(tst, result_jax, result_tf, *, args, tol,
-                      err_msg):  # noqa: F811
-      arg1, arg2 = args
-      # lax.igammac returns 1. when arg1 <= 0; tf.math.igammac returns NaN
-      special_cases = (arg1 <= 0.) | (arg2 <= 0)
-      nr_special_cases = np.count_nonzero(special_cases)
-      tst.assertAllClose(
-          np.full((nr_special_cases,), 1., dtype=dtype),
-          result_jax[special_cases],
-          err_msg=err_msg)
-      tst.assertAllClose(
-          np.full((nr_special_cases,), np.nan, dtype=dtype),
-          result_tf[special_cases],
-          err_msg=err_msg)
-      # non-special cases are equal
-      tst.assertAllClose(
-          result_jax[~special_cases],
-          result_tf[~special_cases],
-          atol=tol,
-          rtol=tol,
-          err_msg=err_msg)
+    def custom_assert_factory(mode):
+      def custom_assert(tst, result_jax, result_tf, *, args, tol,
+                        err_msg, mode=mode):  # noqa: F811
+        arg1, arg2 = args
+        if mode == "compiled":
+          # lax.igammac returns nan. when arg1 <= 0; tf.math.igammac returns 1
+          special_cases = (arg1 <= 0.) | (arg2 <= 0)
+        else:
+          # special_cases will be all False
+          special_cases = (arg1 == np.nan) | (arg2 == np.nan)
```

Comment:
`x == np.nan` will be false for all `x`, including `x = np.nan`. Should we be using `np.isnan`?

Reply:
No. `x == np.nan` never containing True is by design: the special cases tests will be skipped and `nan` content will be compared in the last assert method call.

An alternative approach is to skip the special case tests explicitly (that is, use the full test only when mode is not "compiled") if that is preferred.

Reply:
In that case, I think this is a little too clever for it's own good. I'd be more explicit, like `special_cases = np.zeros(np.broadcast_shapes(arg1.shape, arg2.shape), dtype=bool)`

Reply:
or maybe `np.zeros_like(arg1, dtype=bool) | np.zeros_like(arg2, dtype=bool)` might read better

Reply:
I'll fix it..

Reply:
Done. Basically, the original code is restored and enabling `custom_assert` only when mode is "compiled".